main/inc/lib/myspace.lib.php

```diff
@@ -1368,114 +1466,145 @@ public static function displayResumeLpByItem($startDate = null, $endDate = null,
         $table = '';
         $tblExtraField = Database::get_main_table(TABLE_EXTRA_FIELD);
         $tblExtraFieldValue = Database::get_main_table(TABLE_EXTRA_FIELD_VALUES);
-        $queryExtraFieldPrice = "SELECT id FROM $tblExtraField WHERE variable = 'price'";
-        $queryExtraFieldValue = "SELECT id FROM $tblExtraField WHERE variable = 'authorlpitem'";
-
-        // search items of lp
-        $sql = "SELECT item_id as lp_item_id
-                FROM $tblExtraFieldValue
-                WHERE field_id IN ( $queryExtraFieldValue )
-                GROUP BY lp_item_id";
-        $queryResult = Database::query($sql);
+        $tblLpItem = Database::get_course_table(TABLE_LP_ITEM);
+        $tblLp = Database::get_course_table(TABLE_LP_MAIN);
+        $index = 0;
         $cLpItems = [];
-        while ($row = Database::fetch_array($queryResult, 'ASSOC')) {
-            $cLpItems[] = (int) $row['lp_item_id'];
+        $cLpItemsAuthor = [];
+        $authorArray = [];
+        $studentArray = [];
+        $whereInLpItem = [];
+        $whereInLp = [];
+        $dataSet = [];
+        /** Get lp items only with authors */
+        $sql = " SELECT
+                efv.item_id AS lp_item_id,
+                efv.`value` AS author
+            FROM $tblExtraFieldValue AS efv
+            INNER JOIN $tblExtraField AS ef
+            ON (
+                ef.variable = 'authorlpitem' AND
+                efv.field_id = ef.id and efv.`value` != '' )
+            ORDER BY efv.item_id ";
+        $queryResult = Database::query($sql);
+        $data = Database::store_result($queryResult, 'ASSOC');
+        $totalData = count($data);
+        for ($i = 0; $i < $totalData; $i++) {
+            $cLpItemsAuthor[$data[$i]['lp_item_id']] = $data[$i]['author'];
         }
-
-        if (count($cLpItems) == 0) {
+        /** Get lp items only with price */
+        $sql = " SELECT
+                lpt.iid AS lp_id,
+                lpt.name AS lp_name,
+                efv.item_id AS lp_item_id,
+                lpi.title AS title,
+                efv.`value` AS price
+            FROM $tblExtraFieldValue AS efv
+            INNER JOIN $tblExtraField AS ef
+            ON (
+                ef.variable = 'price' AND
+                efv.field_id = ef.id AND
+                efv.`value` > 0 )
+            INNER JOIN $tblLpItem AS lpi
+            ON ( lpi.iid = efv.item_id )
+            INNER JOIN $tblLp AS lpt
+            ON ( lpi.lp_id = lpt.iid ) ";
+        $queryResult = Database::query($sql);
+        $data = Database::store_result($queryResult, 'ASSOC');
+        $totalData = count($data);
+        for ($i = 0; $i < $totalData; $i++) {
```

Comment:
Usar foreach

Reply:
comentario añadido "use 'for' to performance"